iosHackthonApp/iosHackthonApp/Add Meal Screen/AddMealView.swift

```diff
@@ -8,48 +8,125 @@
 
 import SwiftUI
 import shared
+import CoreLocation
 
 struct AddMealView: View {
-
-    @ObservedObject private(set) var viewModel: AddMealViewModel
-
-    var body: some View {
-        ZStack {}
-        .alert(isPresented: $viewModel.showingCollectionCode) {
-            Alert(
-                title: Text(viewModel.code),
-                message: Text(Strings.AddMealScreen.CollectionAlert.message),
-                dismissButton: .default(Text(Strings.Common.ok)))
-        }
-        ZStack {
-            VStack {
-            //TODO: Pass the actual values when UI is complete
-                Button("Add a meal") {
-                    let meal = Meal(
-                        id: "\(viewModel.sdk.getUUID())",
-                        name: "Pizza",
-                        quantity: 2,
-                        availableFromDate: "Tuesday",
-                        expiryDate: "Saturday",
-                        info: "Meat",
-                        hot: false,
-                        locationLat: 51.509865,
-                        locationLong:  -0.118092)
-                    self.viewModel.postMeal(meal: meal)
-                }
-            }
-        }
-        .alert(isPresented: $viewModel.showingError) {
-            Alert(
-                title: Text(Strings.Common.sorry),
-                message: Text(Strings.Common.ErrorAlert.message),
-                dismissButton: .default(Text(Strings.Common.ok)))
-        }
-    }
+	
+	@ObservedObject private(set) var viewModel: AddMealViewModel
+	@State var title: String = ""
+	@State var additionalInfo: String = ""
+	@State var quantity: Int = 0
+	@State var availableFromDate = Date()
+	@State var useByDate = Date()
+	@State var address: String = ""
+	@State var latitude: Float = 0
+	@State var longitude: Float = 0
+	@State var isHot : Bool = true
```

Comment:
Could these not be refactored into belonging to the viewModel?

Reply:
I am a bit unsure how to get around this since these are State variables.  I have tried doing this but it’s giving me errors.
Happy to tackle this if anyone has any ideas?

Reply:
This isn't my strong suit either.
Maybe try them as @Published properties in the ViewModel

Reply:
Tried, it gives me an error
`Failed to produce diagnostic for expression; please file a bug report`

Reply:
Maybe I m doing something wrong

Reply:
One of them really helpful errors 😖

Reply:
Can we remove the space before the : for `isHot`

Reply:
removed the space 👍